base/mem.c

```diff
@@ -102,6 +106,11 @@ __mem_map_anom(void *base, size_t len, size_t pgsize,
 		  mask ? NNUMA + 1 : 0, MPOL_MF_STRICT | MPOL_MF_MOVE))
 		goto fail;
 
+	if (cfg_transparent_hugepages_enabled && (flags & MAP_HUGETLB)) {
```

Comment:
will flags have the MAP_HUGETLB bit set? maybe we should check for pgsize > PGSIZE_4KB?